### ISSUE TITLE
Error installing

### ISSUE THREAD

**Issue by padpadpadpad** [Bioconductor/Rsamtools#22]
I have an error trying to install Rsamtools using `BiocManager::install('Rsamtools')`. This disrupts the installation of `dada2` for me unfortunately.

```
/bin/sh: 1: Syntax error: "(" unexpected
make: *** [/home2/ISAD/dp415/miniconda3/envs/dada2_pipeline/lib/R/etc/Makeconf:174: Biostrings_stubs.o] Error 2
ERROR: compilation failed for package ‘Rsamtools’
```

Session Info here:

`sessionInfo()
R version 4.0.3 (2020-10-10)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: Ubuntu 18.04.3 LTS

Matrix products: default
BLAS/LAPACK: /home2/ISAD/dp415/miniconda3/envs/dada2_pipeline/lib/libopenblasp-r0.3.10.so

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] BiocManager_1.30.10 compiler_4.0.3      tools_4.0.3        
[4] remotes_2.2.0 `

**Comment by mtmorgan** [comment]
This looks like a problem because of mixed conda / non-conda installations. Have you tried installing Rsamtools using conda? Or alternatively installing all packages outside conda?

**Comment by padpadpadpad** [comment]
Thanks for the quick reply.

So this morning I deleted the conda environment and started again with a fresh R v4.0.3 install. Rsamtools was the first package I attempted to install from inside R and I got the same error.

When using conda (v4.9.2), installing Rsamtools using `mamba install -c bioconda bioconductor-Rsamtools` it requires me to downgrade R to version 3.2.2 which I do not want to do. And seems strange tbh. Any ideas as I would prefer to install everything through conda or everything through the R environment, but errors occur either way.

**Comment by vjcitn** [comment]
I can confirm the problem with mamba
```
        mamba (0.9.0) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['bioconductor-rsamtools=2.6.0']

pkgs/main/noarch         [====================] (00m:00s) Done
pkgs/r/noarch            [====================] (00m:00s) Done
pkgs/r/linux-64          [====================] (00m:00s) Done
bioconda/noarch          [====================] (00m:00s) Done
bioconda/linux-64        [====================] (00m:00s) Done
pkgs/main/linux-64       [====================] (00m:01s) Done
Encountered problems while solving.
Problem: nothing provides libblas >=3.8.0,<4.0a0 needed by bioconductor-rsamtools-2.6.0-r40h5f743cb_0

rstudio@saturn-789a866b-2099-40d8-a7f8-e3855d4fcf81:~$ mamba install -c bioconda bioconductor-rsamtools
```

**Comment by vjcitn** [comment]
I think this is the wrong place to discuss this problem.  It is not an Rsamtools problem.  It is a conda/mamba
problem.  I suspect that some configuration of the conda environment would enable you to have a fully
functional binary install of R 4.0.3 in which you could install Rsamtools using BiocManager.  Please consider
reporting your problem in a bioconda-oriented forum and report back here?  For very confident but possibly tedious
resolution, my prescription would be to use the "shell" of the system on which you are working to install R 4.0.x as a native application, start that R, use `install.packages("BiocManager")` to get a valid version of BiocManager, and then use
`BiocManager::install("Rsamtools")` to get a valid installation of Rsamtools.   Whatever reliance you have on conda environments should be compatible with using this correct version of R/Rsamtools, but the bridging may take a little more work.

**Comment by vjcitn** [comment]
BTW I suspect there are many readers of support.bioconductor.org that have a much better understanding of conda than I do and who would respond to a report of your situation there.

**Comment by padpadpadpad** [comment]
> I think this is the wrong place to discuss this problem. It is not an Rsamtools problem. It is a conda/mamba
> problem. I suspect that some configuration of the conda environment would enable you to have a fully
> functional binary install of R 4.0.3 in which you could install Rsamtools using BiocManager. Please consider
> reporting your problem in a bioconda-oriented forum and report back here? For very confident but possibly tedious
> resolution, my prescription would be to use the "shell" of the system on which you are working to install R 4.0.x as a native application, start that R, use `install.packages("BiocManager")` to get a valid version of BiocManager, and then use
> `BiocManager::install("Rsamtools")` to get a valid installation of Rsamtools. Whatever reliance you have on conda environments should be compatible with using this correct version of R/Rsamtools, but the bridging may take a little more work.

So I tried to do this but I do not have sudo rights for the Linux server, hence my desire to use conda and have everything inside an environment... Maybe Docker is the long-term solution to this but seems like overkill.

**Comment by vjcitn** [comment]
You don't need sudo rights to install R.  This predicament is pretty common I would say and if you articulated it to support.bioconductor.org I think you would get some very thoughtful and germane responses.

**Comment by mtmorgan** [comment]
For what it's worth I did
```
docker run -it --rm bioconductor/bioconductor_docker:RELEASE_3_12 bash
```
and then
```
root@d70ae87ef418:/# R -e "BiocManager::install('Rsamtools')"
```
which completed successfully. This shows that it is something local to your configuration. Also when you said you started again and 'got the same error message' I guess that's not exactly true, because the error message mentions conda... So what was the exact error? If this is an HPC cluster, then it might be worth while reaching out to the server administrator -- these people often like solving weird problems like this, maybe for the same reason writing software can be fun -- you introduce a bug, then solve it and whoever the bug was causing problems for thinks you're amazing.

**Comment by padpadpadpad** [comment]
Thanks for doing this. I tried to delete my miniconda installation and start again. i somehow got some processes stuck so could only try again just now after the server was reset.

It now works, I think the problem could have been something in my `.Renviron` file. Specifically previously `export ASREML_LICENSE_FILE` was in it and caused warnings in R package installs. I have now managed to install the correct version of R and all packages install successfully from within R.
